docs/testing/playwright/02-e2e-testing.md

```diff
@@ -0,0 +1,125 @@
+---
+title: End-to-End Testing
+sidebar_label: E2E Testing
+---
+
+When it comes to writing end-to-end tests using the Stencil Playwright adapter, the best advice we can give is to leverage
+[Playwright documentation](https://playwright.dev/docs/writing-tests). The adapter is set up in a way so that developers will use the
+same public APIs with as little Stencil nuances as possible.
+
+## Writing Tests
+
+As far as writing the actual tests goes, the most important thing to be aware of for tests to run correctly is to import the `test` function from
+`@stencil/playwright`, **not** directly from `@playwright/test`:
+
+```ts
+// THIS IS CORRECT
+import { test } from '@stencil/playwright';
+
+// THIS IS NOT!!
+// import { test } from '@playwright/test';
+```
+
+The adapter package extends Playwright's stock `test` function to provide additional fixtures and handle nuances related to web component hydration. More
+information is available in the [API documentation](./03-api.md#test-function).
+
+### Testing Patterns
+
+#### `page.goto()`
+
+The `goto()` method allows tests to load a pre-defined HTML template. This pattern is great if a test file has many tests to execute that all use the same HTML code
+or if additional `script` or `style` tags need to be included in the HTML. However, with this pattern, developers are responsible for defining the necessary `script`
+tags pointing to the Stencil entry code (so all web components are correctly loaded and registered).
+
+```html title="my-component.e2e.html"
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf8" />
+
+    <!-- Replace with the path to your entrypoint -->
+    <script src="./build/test-app.esm.js" type="module"></script>
+    <script src="./build/test-app.js" nomodule></script>
```

Comment:
Are these paths relative to `www`? I'm having trouble getting a 'hello world' style test setup using this example here, and suspect it's because I'm not loading these correctly.

For a starter created with `npm init stencil@latest component playwright-test`:
```html
<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf8" />

    <!-- Replace with the path to your entrypoint -->
    <script src="./wbuild/playwright-test.esm.js" type="module"></script>
    <script src="./build/playwright-test.js" nomodule></script>
  </head>
  <body>
    <my-component first="Stencil"></my-component>
  </body>
</html>
```
and 
```ts
import {expect} from '@playwright/test'
import {test} from '@stencil/playwright';

test.describe('my-component', () => {
  test('should render the correct name', async ({page}) => {
    await page.goto('my-component.e2e.html');
    expect(true).toBe(true)
  })
})
```

This times out after 30 secounds on the `goto`:
```
  1) src/components/my-component/my-component.e2e.ts:5:7 › my-component › should render the correct name

    Test timeout of 30000ms exceeded.

    Error: page.waitForFunction: Test timeout of 30000ms exceeded.

      4 | test.describe('my-component', () => {
      5 |   test('should render the correct name', async ({page}) => {
    > 6 |     await page.goto('/my-component/my-component.e2e.html');
        |                ^
      7 |     expect(true).toBe(true)
      8 |   })
      9 | })
```

Is there anything I'm missing that's blatantly obvious here? If I run this with `--headed` mode enabled, I get
```
404 File Not Found
Url: /my-component/my-component.e2e.html
File: /private/tmp/playwright-test/www/my-component/my-component.e2e.html
```
where the location of `my-component.e2e.html` on disk is actually `/private/tmp/playwright-test/src/components/my-component/my-component.e2e.html` - should my HTML file be under `www`?

Reply:
Good call out. Totally meant to try and clarify and guide a bit more here cuz this is probably the trickiest bit in the whole adapter. I'll throw something together

Reply:
Added some text to clarify the expected values from the code snippets. Let me know if that is clear enough or if there is more I can add.